gpt_index/tts/base.py

```diff
@@ -0,0 +1,67 @@
+"""Text to speech module."""
+
+from typing import Optional
+
+
+class barkTTS:
+    def __init__(self, text_temp: float = 0.7, waveform_temp: float = 0.7) -> None:
+        """
+        Args:
+            text_temp: generation temperature (1.0 more diverse, \
+                0.0 more conservative)
+            waveform_temp: generation temperature (1.0 more diverse, \
+                0.0 more conservative)
+        """
+
+        import_err_msg = "`bark` package not found, \
+            please run `pip install git+https://github.com/suno-ai/bark.git`"
+        try:
+            import bark
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        self.generate_audio = bark.generate_audio
+        self.save_as_prompt = bark.save_as_prompt
+        self.text_temp = text_temp
+        self.waveform_temp = waveform_temp
+
+    def generate_bark_audio(
+        self, text: Optional[str], lang_speaker_voice: Optional[str]
+    ) -> None:
```

Comment:
would be good to abstract out a BaseTTS class too that takes in text as a single argument. lang_speaker_voice could probably be moved to the `__init__`

Reply:
also lang_speaker_voice could be an optional kwarg